### DISCUSSION TITLE
Castilian vs Spanish

### DISCUSSION THREAD

**Discussion by lokesh12345678910** [openai/whisper#1793] [Q&A]
I've transcribed in Castillian through `whisper –model large-v2 –language Castilian`. Does this have the same output as --language Spanish?

I asked the owner of a different package to add the Whisper Castilian model to their package. This was their response: 

> I haven’t found a separate model that supports Castilian Spanish; the appendix of Whisper’s paper ([https://cdn.openai.com/papers/whisper.pdf](https://urldefense.com/v3/__https:/cdn.openai.com/papers/whisper.pdf__;!!BuQPrrmRaQ!gf4LMpYMbnbiWGHiiEuANGNRlGP1tnAiaBLCUiTpd7Rvi6P7tMPofZ7qpRBE9zFRKtVBV4zf3YfpE-ckXFujXKp1ZPnR$)) only lists “Spanish” as a language they had data for. 
> 
> Perhaps Castilian is an accepted flag that also means Spanish? Have you found that supplying “Castilian” has performance differences on the Whisper implementation you are using?>

Is there a seperate model that supports Castilian?

**Comment by glangford** [comment]
> Is there a seperate model that supports Castilian?

In the current version of whisper, Castilian appears to be aliased to Spanish.

https://github.com/openai/whisper/blob/fcfeaf1b61994c071bba62da47d7846933576ac9/whisper/tokenizer.py#L126

**Reply by lokesh12345678910** [reply]
Has there been any version of whisper (large-v2 or large-v1) where Castilian isn't aliased to Spanish?

**Reply by glangford** [reply]
> Has there been any version of whisper (large-v2 or large-v1) where Castilian isn't aliased to Spanish?

It looks like the alias to Spanish goes back to the original version of whisper, so I think it has been this way for all models from the beginning.

**Comment by sephethus** [comment]
Castilian is very distinctly different than American Spanish, more so than British English is from American. I don't understand why this isn't well understood.
